samcli/commands/validate/lib/sam_template_validator.py

```diff
@@ -136,7 +94,7 @@ def is_s3_uri(uri):
             Returns True if the uri given is an S3 uri, otherwise False
 
         """
-        return isinstance(uri, str) and uri.startswith("s3://")
+        return external_is_s3_uri(uri)
```

Comment:
will there be any chance that the input uri is a path style s3 uri? e.g. `https://s3.amazonaws.com/bucket/key` or `https://s3-us-east-2.amazonaws.com/bucket/key`?

Reply:
I'm not entirely sure. The only changes I made to sam_template_validator is I extracted the code logic into another class so that I could use it in `sam check`.